src/compile/data/calculate.ts

```diff
@@ -15,6 +19,30 @@ export class CalculateNode extends DataFlowNode {
     super(parent);
   }
 
+  public static parseAllForSortIndex(parent: DataFlowNode, model: ModelWithField) {
+    // get all the encoding with sort fields from model
+    model.forEachFieldDef((fieldDef: ScaleFieldDef<string>, channel: SingleDefChannel) => {
+      if (isScaleFieldDef(fieldDef) && isSortArray(fieldDef.sort)) {
+        const transform: CalculateTransform = {
+          calculate: CalculateNode.calculateExpressionFromSortField(fieldDef.field, fieldDef.sort),
+          as: sortArrayIndexField(model, channel)
```

Comment:
It's weird that you're still using `model` and `channel` given that this is already under `model.forEachFieldDef`.  


I'd say `sortArrayIndexField` should just accept a `fieldDef`

Reply:
If that's the case, how can we know what the channel the given `fieldDef` represents?

Reply:
I think we should still leave `channel` as function param

Reply:
Sorry for the long reply, since this function is also used in `domain.ts` to look up the sort field, given that only `model` and `channel` is provided in that code, I think accepting `model` and `channel` here is reasonable, even though it seems redundant in `model.forEachFieldDef`.